src/internals/commandHandler.ts

```diff
@@ -49,47 +64,58 @@ function startCommandHandler(client: Client): Collection<string, Command> {
 			return;
 		}
 		for (const [key, val] of commandData) {
-			const commandPermissions = commands.get(val.name)?.permissions;
-			if (!commandPermissions) continue;
+			const command = commands.get(val.name);
+			if (!command || !command.permissions) continue;
 			const data: GuildApplicationCommandPermissionData = {
 				id: key,
-				permissions: commandPermissions
+				permissions: command.permissions
 			};
 			permissions.push(data);
 			const permSetInfo = await guild.commands.permissions.set({ fullPermissions: permissions });
 			if (permSetInfo.size === 0) {
-				console.log("No permissions have been set");
+				console.log("No permissions have been set.");
 				return;
 			}
-			console.log("Permissions set successfully");
+			console.log("Permissions set successfully.");
 			return;
 		}
 	}
 
 	client.once("ready", async () => {
 		// Loading commands from /commands.
-		await loadFiles<Command>(commands, join(__dirname, "..", "commands"));
-		const s = commands.size === 1 ? "" : "s";
-		console.log(`${commands.size} command${s} loaded.`);
+		await loadFiles<SlashCommand>(slashCommands, join(__dirname, "..", "commands"));
+		await loadFiles<ContextMenuCommand>(contextCommands, join(__dirname, "..", "context-commands"));
 
-		// Setting the commands as slash commands in the selected guild.
-		commandData = await setCommands(commands);
+		console.log(`${slashCommands.size} slash command${plural(slashCommands.size)} loaded.`);
+		console.log(`${contextCommands.size} context menu item${plural(contextCommands.size)} loaded.`);
 
-		// Configuring permissions for every command.
-		await setPermissions(commands, commandData);
+		if (!process.argv.includes("--no-edit")) {
+			// Setting the commands as slash commands in the selected guild.
+			commandData = await setCommands(slashCommands, contextCommands);
+
+			// Configuring permissions for every command.
+			await setPermissions(slashCommands, commandData);
+		} else {
+			console.log("No commands set.");
+		}
 	});
 	client.on("interactionCreate", (interaction) => {
-		if (!interaction.isCommand()) {
-			return;
-		}
-		const cmd = commands.get(interaction.commandName);
-		if (!cmd) {
-			return;
+		if (interaction.isCommand()) {
```

Comment:
is it possibel to use something like `slashCommands.get(interaction.commandname)?.execute(interaction)`?
same goes for line 110

Reply:
This format is more suitable for future internal checks.